From d95cf569c17f8d2d68029cba6cdbfc6c26924256 Mon Sep 17 00:00:00 2001
From: Yahan Lu <yahan@iscas.ac.cn>
Date: Fri, 9 Apr 2021 13:56:43 +0800
Subject: [PATCH] [riscv64][wasm][liftoff] Record correct offset in
 StoreTaggedPointer

Port: a1616e6f7f3ad60fe426206afefab729ed095b49
Change-Id: Idfb48da2e38948b23efdc129da8949200f0896c3
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/2814723
Commit-Queue: Brice Dobry <brice.dobry@futurewei.com>
Reviewed-by: Brice Dobry <brice.dobry@futurewei.com>
Reviewed-by: Jakob Gruber <jgruber@chromium.org>
Cr-Commit-Position: refs/heads/master@{#74184}
---
 src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h | 2 +-
 test/mjsunit/mjsunit.status                           | 4 ----
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h b/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h
index f2b698685b..bc1ec79258 100644
--- a/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h
+++ b/src/wasm/baseline/riscv64/liftoff-assembler-riscv64.h
@@ -452,7 +452,7 @@ void LiftoffAssembler::StoreTaggedPointer(Register dst_addr,
   JumpIfSmi(src.gp(), &exit);
   CheckPageFlag(src.gp(), scratch,
                 MemoryChunk::kPointersToHereAreInterestingMask, eq, &exit);
-  Add64(scratch, dst_addr, offset_imm);
+  Add64(scratch, dst_op.rm(), dst_op.offset());
   CallRecordWriteStub(dst_addr, scratch, EMIT_REMEMBERED_SET, kSaveFPRegs,
                       wasm::WasmCode::kRecordWrite);
   bind(&exit);
diff --git a/test/mjsunit/mjsunit.status b/test/mjsunit/mjsunit.status
index 258b391258..d8ab20ef3b 100644
--- a/test/mjsunit/mjsunit.status
+++ b/test/mjsunit/mjsunit.status
@@ -854,10 +854,6 @@
 
 }],  # 'arch == riscv64'
 
-['arch == riscv64 and variant == stress_incremental_marking', {
-  # https://github.com/v8-riscv/v8/issues/414
-  'wasm/externref-globals-liftoff': [SKIP],
-}], #'arch == riscv64 and variant == stress-incremental-marking'
 
 ##############################################################################
 ['system == macos', {
-- 
2.35.1

